### PR TITLE
chore: adaptations for nightly-2024-08-14

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -972,6 +955,7 @@ theorem indexOf_inj [DecidableEq α] {l : List α} {x y : α} (hx : x ∈ l) (hy
       simp only [h]
     simp only [indexOf_get] at x_eq_y; exact x_eq_y, fun h => by subst h; rfl⟩
 
+@[deprecated getElem_reverse_aux₂ (since := "2024-08-15")]
 theorem getElem_reverse_aux₂ :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/List/Basic.lean:959]: Is this deprecated in favor of the same lemma?

File: Mathlib/Data/List/Infix.lean [hunk 2]
```diff
@@ -35,36 +33,23 @@ variable {l l₁ l₂ l₃ : List α} {a b : α} {m n : ℕ}
...
-  prefix_refl _
+  prefix_refl
 
+@[deprecated suffix_refl (since := "2024-08-15")]
 theorem suffix_rfl : l <:+ l :=
-  suffix_refl _
+  suffix_refl
 
+@[deprecated infix_refl (since := "2024-08-15")]
 theorem infix_rfl : l <:+: l :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/List/Infix.lean:47]: Wait, is upstream using `_refl` for the implicit argument version? That should be fixed

File: Mathlib/Data/List/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Data/List/Basic.lean:940]: Oops, should just be deprecated without a suggestion: this is just an auxiliary lemma for a now deprecated theorem, `get_reverse`.

File: Mathlib/Data/List/Infix.lean (hunk 2 above)
Comment (semorrison) [reply Mathlib/Data/List/Infix.lean:47]: Ah, indeed. I've removed the deprecations and will fix upstream.

Comment (semorrison) [reply Mathlib/Data/List/Infix.lean]: https://github.com/leanprover/lean4/pull/5077

**Comment by jcommelin** [comment]
bors d+

**Pull request merged**